pinot-controller/src/main/java/com/linkedin/pinot/controller/ControllerConf.java

```diff
@@ -56,6 +59,12 @@
   // protection is enabled. If the upload does not finish within the timeout, next upload can override the previous one.
   private static final String SEGMENT_UPLOAD_TIMEOUT_IN_MILLIS = "controller.segment.upload.timeoutInMillis";
 
+  // Defines the kind of storage and the underlying PinotFS implementation;
+  // each storage type will have a different factory class with this prefix
```

Comment:
Is that true? I think we will have one factory class that will look up the config to instatiate the right FS class.